addons/mail/controllers/bus.py

```diff
@@ -43,7 +43,7 @@ def mail_chat_post(self, uuid, message_content, **kwargs):
         # post a message without adding followers to the channel. email_from=False avoid to get author from email data
         mail_channel = request.env["mail.channel"].sudo().search([('uuid', '=', uuid)], limit=1)
         body = tools.plaintext2html(message_content)
-        message = mail_channel.sudo().with_context(mail_create_nosubscribe=True).message_post(author_id=author_id, email_from=False, body=body, message_type='comment', subtype='mail.mt_comment')
+        message = mail_channel.with_user(request.env.ref('base.user_root')).with_context(mail_create_nosubscribe=True).message_post(author_id=author_id, email_from=False, body=body, message_type='comment', subtype='mail.mt_comment')
```

Comment:
What I would suggest

  * if uid -> author_id = uid, email_from = author.email_formatted based on uid
  * if no uid -> take info from channel create_uid.company_id.catchall, which should give you an email and a reply-to valid that does not allow spam (writing to catchall is not permitted)